mintapi/api.py

```diff
@@ -163,46 +159,92 @@ def get_email_code(imap_account, imap_password, imap_server, imap_folder, debug=
+        version=version, arch=CHROME_ZIP_TYPES.get(arch))
+
+
+def get_chrome_driver_major_version_from_executable(local_executable_path):
+    # Note; --version works on windows as well.
+    version = subprocess.check_output([local_executable_path, '--version'])
+    return version_pattern.match(version.decode()).groupdict()['major']
+
+
+def get_latest_chrome_driver_version():
+    """Returns the version of the latest stable chromedriver release."""
+    latest_url = CHROME_DRIVER_BASE_URL + CHROME_DRIVER_LATEST_RELEASE
+    latest_request = requests.get(latest_url)
+
+    if latest_request.status_code != 200:
+        raise RuntimeError(
+            'Error finding the latest chromedriver at {}, status = {}'.format(
+                latest_url, latest_request.status_code))
+    return latest_request.text
+
+
+def get_stable_chrome_driver():
+    chromedriver_name = 'chromedriver'
+    if _platform in ['win32', 'win64']:
+        chromedriver_name += '.exe'
+
+    local_executable_path = os.getcwd() + os.path.sep + chromedriver_name
+
+    latest_chrome_driver_version = get_latest_chrome_driver_version()
+    latest_major_version = version_pattern.match(
+        latest_chrome_driver_version).groupdict()['major']
+    if os.path.exists(local_executable_path):
+        major_version = get_chrome_driver_major_version_from_executable(
+            local_executable_path)
+        if major_version == latest_major_version:
+            return local_executable_path
+        logger.info('Removing old version {} of Chromedriver'.format(
+            major_version))
+        os.remove(local_executable_path)
+
+    logger.info('Downloading version {} of Chromedriver'.format(
+        latest_chrome_driver_version))
+    zip_file_url = get_chrome_driver_url(
+        latest_chrome_driver_version, _platform)
+    request = requests.get(zip_file_url)
+
+    if request.status_code != 200:
+        raise RuntimeError(
+            'Error finding chromedriver at {}, status = {}'.format(
+                zip_file_url, request.status_code))
+
+    zip_file = zipfile.ZipFile(io.BytesIO(request.content))
+    zip_file.extractall()
+    os.chmod(local_executable_path, 0o755)
+    return local_executable_path
 
 
 def get_web_driver(email, password, headless=False, mfa_method=None,
                    mfa_input_callback=None, wait_for_sync=True,
                    wait_for_sync_timeout=5 * 60,
                    session_path=None, imap_account=None, imap_password=None,
-                   imap_server=None, imap_folder="INBOX"):
+                   imap_server=None, imap_folder="INBOX",
+                   use_chromedriver_on_path=False):
     if headless and mfa_method is None:
-        warnings.warn("Using headless mode without specifying an MFA method"
-                      "is unlikely to lead to a successful login. Defaulting --mfa-method=sms")
+        logger.warn("Using headless mode without specifying an MFA method"
```

Comment:
Will logger.warn go to the terminal like `warnings`?

Reply:
Yes; if you read up on warnings, the intent is to make it easy to filter out based on type of warning. However, mintapi did not use any of their structured warnings to make use of this feature. In this case, it's basically identical, and due to the log level being at INFO (warning is more severe than info), all of the .warn calls will go to stderr (and also INFOs). The [StreamHandler](https://docs.python.org/2/library/logging.handlers.html#logging.StreamHandler) sets this up.